examples/simple_login.rs

```diff
@@ -0,0 +1,203 @@
+//! Demonstrates a simple client-server password-based login protocol
+//! using OPAQUE, over a command-line interface
+
+use rand_core::OsRng;
+use rustyline::error::ReadlineError;
+use rustyline::Editor;
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::process::exit;
+
+use opaque_ke::{
+    ciphersuite::CipherSuite,
+    keypair::KeyPair,
+    opaque::{
+        ClientLogin, ClientRegistration, LoginFirstMessage, LoginSecondMessage, LoginThirdMessage,
+        RegisterFirstMessage, RegisterSecondMessage, RegisterThirdMessage, ServerLogin,
+        ServerRegistration,
+    },
+};
+
+// The ciphersuite trait allows to specify the underlying primitives
```

Comment:
How about an initial comment to describe the point of the example above here?

Reply:
Isn't that already taken care of by the top-level comment at the top of this file?

Reply:
I was expecting one more line on the degree of variability you can and should expect in your implementation. Here are things that could improve what we have now:
- where the client/server boundary is w.r.t the CLI, and in particular, where is the metaphorical "wire", and does the "get_username_and_password" function run on one side or the other?
- you can change all elements in the CipherSuite, but not in-between sub-flows
- each sub-flow has 3-steps, that must be performed in a specific sequence